build-gradle/build.sh

```diff
@@ -0,0 +1,238 @@
+  local pull_request="${PULL_REQUEST:-false}"
+
+  if [[ -z "${SONAR_HOST_URL:-}" || -z "${SONAR_TOKEN:-}" ]]; then
+    return 0
+  fi
+
+  if [[ "$GITHUB_REF_NAME" == "master" && "$pull_request" == "false" ]]; then
+    # Master branch analysis
+    args_ref+=("-Dsonar.projectVersion=$INITIAL_VERSION")
+    args_ref+=("-Dsonar.analysis.sha1=$GITHUB_SHA")
+
+  elif [[ "$GITHUB_REF_NAME" == branch-* && "$pull_request" == "false" ]]; then
+    # Maintenance branch analysis
+    args_ref+=("-Dsonar.branch.name=$GITHUB_REF_NAME")
+    args_ref+=("-Dsonar.projectVersion=$INITIAL_VERSION")
+    args_ref+=("-Dsonar.analysis.sha1=$GITHUB_SHA")
+
+  elif [[ "$pull_request" != "false" ]]; then
+    # Pull request analysis
+    args_ref+=("-Dsonar.analysis.sha1=$PULL_REQUEST_SHA")
+    args_ref+=("-Dsonar.analysis.prNumber=$PULL_REQUEST")
+
+  elif [[ "$GITHUB_REF_NAME" == feature/long/* && "$pull_request" == "false" ]]; then
+    # Long-lived feature branch analysis
+    args_ref+=("-Dsonar.branch.name=$GITHUB_REF_NAME")
+    args_ref+=("-Dsonar.analysis.sha1=$GITHUB_SHA")
+  fi
+}
+
+gradle_build() {
+  # Setup Gradle
+  if command_exists gradle; then
+    GRADLE_CMD="gradle"
+  elif [[ -f "./gradlew" ]]; then
+    GRADLE_CMD="./gradlew"
+    chmod +x ./gradlew
+  else
+    echo "Neither gradle nor gradlew found" >&2
+    return 1
+  fi
+
+  echo "Using Gradle command: $GRADLE_CMD"
+  export GRADLE_CMD
+
+  # Conditional validation for SonarQube
+  if [[ -n "${SONAR_HOST_URL:-}" && -z "${SONAR_TOKEN:-}" ]]; then
+    echo "SONAR_TOKEN is required when SONAR_HOST_URL is set" >&2
+    return 1
+  fi
+
+  # Conditional validation for Artifactory
+  if [[ -n "${ARTIFACTORY_DEPLOY_REPO:-}" && ( -z "${ARTIFACTORY_DEPLOY_USERNAME:-}" || -z "${ARTIFACTORY_DEPLOY_PASSWORD:-}" ) ]]; then
+    echo "ARTIFACTORY_DEPLOY_USERNAME and ARTIFACTORY_DEPLOY_PASSWORD are required when ARTIFACTORY_DEPLOY_REPO is set" >&2
+    return 1
+  fi
+
+  local gradle_args
+  read -ra gradle_args <<< "$(build_gradle_args)"
+
+  set_sonar_args gradle_args
+
+  local build_type
+  local pull_request="${PULL_REQUEST:-false}"
+
+  if [[ "$GITHUB_REF_NAME" == "master" && "$pull_request" == "false" ]]; then
+    build_type="master branch"
+  elif [[ "$GITHUB_REF_NAME" == branch-* && "$pull_request" == "false" ]]; then
+    build_type="maintenance branch"
+  elif [[ "$pull_request" != "false" ]]; then
+    build_type="pull request"
+  elif [[ "$GITHUB_REF_NAME" == dogfood-on-* && "$pull_request" == "false" ]]; then
+    build_type="dogfood branch"
+  elif [[ "$GITHUB_REF_NAME" == feature/long/* && "$pull_request" == "false" ]]; then
+    build_type="long-lived feature branch"
+  else
+    build_type="regular build"
+  fi
+
+  echo "Starting $build_type build..."
+  echo "Gradle command: $GRADLE_CMD ${gradle_args[*]}"
+
+  if "$GRADLE_CMD" "${gradle_args[@]}"; then
+    echo "Build completed successfully"
+    return 0
+  else
+    echo "Build failed" >&2
+    return 1
+  fi
```

Comment:
Not needed and hiding real error code.
`"$GRADLE_CMD" "${gradle_args[@]}"` will return success or failure by itself.